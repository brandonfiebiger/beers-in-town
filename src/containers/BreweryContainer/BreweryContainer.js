import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BreweryCard } from '../BreweryCard/BreweryCard'


export class BreweryContainer extends Component {
  constructor(props) {
    super(props);
    
  }
  
  displayBreweries = () => this.props.breweries.map(brewery => {
      return <BreweryCard {...brewery}/>
    })
  

  
  render() {
    return(
      <section>
        {this.displayBreweries()}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  breweries: state.breweries
})



export default connect(mapStateToProps, null)(BreweryContainer);


