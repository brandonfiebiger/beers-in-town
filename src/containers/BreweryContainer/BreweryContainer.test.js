import { BreweryContainer } from './BreweryContainer';
import { shallow, mount } from 'enzyme';
import React from 'react';

describe('BreweryContainer', () => {

  let wrapper;

  beforeEach(() => {

    wrapper = shallow(<BreweryContainer breweries={[]}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with breweries', () => {
    wrapper = shallow(<BreweryContainer breweries={[{name: 'Boo'}]}/>)

    expect(wrapper).toMatchSnapshot()

  })
})