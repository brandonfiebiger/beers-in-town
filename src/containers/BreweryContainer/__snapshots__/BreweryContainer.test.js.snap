// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BreweryContainer
    breweries={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`BreweryContainer should match the snapshot with breweries 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BreweryContainer
    breweries={
        Array [
            Object {
              "name": "Boo",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BreweryCard
          name="Boo"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "Boo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BreweryCard
            name="Boo"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "Boo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
